Makefile

```diff
@@ -8,268 +28,405 @@ usage:
-	@echo "            list: Lists available boards"
+	@echo
+	@echo "The make system also drives all continuous integration and testing:"
+	@echo "         $$(tput bold)prepush$$(tput sgr0): Fast checks to run before pushing changes upstream"
+	@echo "          ci-all: Run all continuous integration tests (possibly slow!)"
+	@echo "         ci-help: More information on Tock CI and testing"
 	@echo
 	@echo "$$(tput bold)Happy Hacking!$$(tput sgr0)"
 
+##
+## End: usage.
+##
+################################################################################
+##
+## Utility functions
+##
+
+define banner
+	@printf "\n"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@string="$(1)" && printf "$$(tput bold)* %-$$((76))s *\n" "$$string"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@printf "\n"
+endef
+
+##
+## End: functions.
+##
+################################################################################
+##
+## User convenience targets
+##
+
+# Aggregate targets
+.PHONY: allaudit
+audit:
+	@for f in `./tools/list_lock.sh`;\
+		do echo "$$(tput bold)Auditing $$f";\
+		(cd "$$f" && cargo audit || exit 1);\
+		done
+
 .PHONY: allboards
 allboards:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Build $$f"; $(MAKE) -C "boards/$$f" || exit 1; done
+	@for f in $(ALL_BOARDS);\
+		do echo "$$(tput bold)Build $$f";\
+		$(MAKE) -C "boards/$$f" || exit 1;\
+		done
 
 .PHONY: allcheck
 allcheck:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Check $$f"; $(MAKE) -C "boards/$$f" check || exit 1; done
+	@for f in $(ALL_BOARDS);\
+		do echo "$$(tput bold)Check $$f";\
+		$(MAKE) -C "boards/$$f" check || exit 1;\
+		done
 
 .PHONY: alldoc
 alldoc:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Documenting $$f"; $(MAKE) -C "boards/$$f" doc || exit 1; done
+	@for f in $(ALL_BOARDS);\
+		do echo "$$(tput bold)Documenting $$f";\
+		$(MAKE) -C "boards/$$f" doc || exit 1;\
+		done
 
 
+# Commands
+.PHONY: fmt format
+fmt format:
+	@./tools/run_cargo_fmt.sh
 
-###################################################################
+.PHONY: clean
+clean:
+	@echo "$$(tput bold)Clean top-level Cargo workspace" && cargo clean
+	@for f in `./tools/list_tools.sh`;\
+		do echo "$$(tput bold)Clean tools/$$f";\
+		cargo clean --manifest-path "tools/$$f/Cargo.toml" || exit 1;\
+		done
+	@echo "$$(tput bold)Clean rustdoc" && rm -rf doc/rustdoc
+	@echo "$$(tput bold)Clean ci-artifacts" && rm -rf ./ci-artifacts
+
+
+## Meta-Targets
+
+# Run all possible CI. If this passses locally, all cloud CI *must* pass as well.
+.PHONY: ci-all
+ci-all:\
+	ci-runner-github\
+	ci-runner-netlify
+
+# Run the fast jobs.
+# This is designed for developers, to be run often and before submitting code upstream.
+.PHONY: prepush
+prepush:\
+	ci-job-format\
```

Comment:
I think it's intended that prepush only performs checks but doesn't edit any files, right?

Reply:
That was the thought.., but really why would you as a developer want to read diffs at this stage? Especially because the `format` job does check that all changes are at least staged before writing any files (i.e. it's easy to undo its actions). I'll change this.